### PR TITLE
resolves #1163 debugF function is not respected

### LINKED ISSUE #1163: `Options.Debugf` is not used properly
### Describe the bug
I've set the custom `Debugf` function in `clickhouse.Options` struct, but it is not used.

### Steps to reproduce
1. Use `clickhouse.Options{Debug: true, Debugf: func(format string, v ...any) {  }}`
2. Watch some logging info appear even though it should not (easy to test on `Rows Close error: context canceled`)

#### Environment
* Client version: v2.16.0
* Language version: go1.21.5 linux/amd64
* OS: Fedora release 39
* Interface: database/sql compatible driver

### PR DESCRIPTION
**Pull request by omurbekjk**
## Summary

Options.DebugF is not used properly.
Resolves #1163 

## Checklist
Delete items not relevant to your PR:
- [X] Unit and integration tests covering the common scenarios were added
- [X] A human-readable description of the changes was provided to include in CHANGELOG
- [X] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: tests/issues/1163_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	port := env.Port
+	if useSSL {
+		tlsConfig = &tls.Config{}
+		port = env.SslPort
+	}
+	options := &clickhouse.Options{
+		Addr:  []string{fmt.Sprintf("%s:%d", env.Host, port)},
+		Debug: true,
+		Debugf: func(format string, v ...any) {
+
```
Comment (jkaflik) [inline tests/issues/1163_test.go:30]: it's worth adding a flag here, so it's assert later
```suggestion
debugfCalled = true
```

File: tests/issues/1163_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			Database: "default",
+			Username: env.Username,
+			Password: env.Password,
+		},
+		TLS: tlsConfig,
+	}
+	conn := clickhouse.Connector(options)
+	c, err := conn.Connect(context.TODO())
+	require.NoError(t, err)
+	require.NotNil(t, c)
```
Comment (jkaflik) [inline tests/issues/1163_test.go:42]:
```suggestion
	require.NotNil(t, c)
	require.True(debugfCalled)
```

**Review by jkaflik — COMMENTED** [review]
@omurbekjk thanks! would you mind signing CLA?

File: tests/issues/1163_test.go (hunk 1 above)
Comment (omurbekjk) [reply tests/issues/1163_test.go]: thanks for reviewing. Addressed MR comment

**Review by jkaflik — APPROVED** [review]

**Pull request merged**